core/src/main/java/com/linecorp/armeria/common/ContentTooLargeException.java

```diff
@@ -23,20 +23,72 @@
 
     private static final long serialVersionUID = 4901614315474105954L;
 
+    private long transferred;
+    private long transferring;
+    private long maximum;
+
     private static final ContentTooLargeException INSTANCE = new ContentTooLargeException(false);
 
     /**
      * Returns a {@link ContentTooLargeException} which may be a singleton or a new instance, depending on
      * {@link Flags#verboseExceptionSampler()}'s decision.
+     * @deprecated use {@link #builder()} instead.
      */
     public static ContentTooLargeException get() {
         return Flags.verboseExceptionSampler().isSampled(ContentTooLargeException.class) ?
                new ContentTooLargeException() : INSTANCE;
     }
 
+    /**
+     * Returns a {@link ContentTooLargeExceptionBuilder} which may return a singleton or a new instance in its
+     * {@link ContentTooLargeExceptionBuilder#build()} method, depending on
+     * {@link Flags#verboseExceptionSampler()}'s decision.
+     */
+    public static ContentTooLargeExceptionBuilder builder() {
+        return new ContentTooLargeExceptionBuilder();
+    }
+
     private ContentTooLargeException() {}
 
     private ContentTooLargeException(@SuppressWarnings("unused") boolean dummy) {
         super(null, null, false, false);
     }
```

Comment:
Two things:

We could set all private member fields in these two constructors, and then we can make all of them `final`.

```suggestion
    private final boolean neverSample;

    private ContentTooLargeException() {
        this(false);
    }

    private ContentTooLargeException(boolean neverSample) {
        super(null, null, !neverSample, !neverSample);
        this.neverSample = neverSample;
    }
```

Please also check my comment below at `fillInStackTrace()`.